### ISSUE TITLE
Mob#clearLootTable has no effect

### ISSUE THREAD

**Issue by NonSwag** [PaperSpigot/Paper#11849]
### Expected behavior

`Mob#clearLootTable` (or `setLootTable(null)`) should prevent the mob from generating and dropping loot on death

### Observed/Actual behavior

The mob still generates and drops its default loot

### Steps/models to reproduce

1. spawn a mob (pig for example)
2. invoke the `Mob#clearLootTable` method (or set it to null)
3. kill the entity

### Plugin and Datapack List

irrelevant

### Paper version

This server is running Paper version 1.21.4-61-main@e0abc5f (2024-12-28T14:03:42Z) (Implementing API version 1.21.4-R0.1-SNAPSHOT)
You are running the latest version
Previous version: 1.21.4-34-d568e17 (MC: 1.21.4)

**Comment by Warriorrrr** [comment]
This is due to the way Mob#getLootTable works in internals, it defaults to the entity type's default loot table if the Mob#lootTable field is an empty optional, which is what is set by the API.

Though what you're trying to do is already possible in the API using the EntityDeathEvent, just clear the getDrops there and the entity won't drop anything

**Comment by NonSwag** [comment]
Yeah, I just came to the same conclusion while reading through the code.
for now, I just pass an empty loot table instance because an event isn't a solution in my case (the loot table might get overridden later in the entity's lifecycle).

Judging from the docs, it sounds like setting the loot table to null will not generate loot on death of a mob.

![Image](https://github.com/user-attachments/assets/3dbbec39-680b-420e-b0da-0ddb10d67622)

I feel like adding a `resetLootTable` method would be clearer for that, while also making it possible to have no loot table at all.
Also, getting the loot table after setting it to null, on an entity that has a loot table by default, will not return null which seems wrong.

**Comment by electronicboy** [comment]
This is pretty much the joys of making loose promises in the API over existing implementations of behavior, if we want to enforce this API promise, most solutions here just introduce their own behavioral breakages, I think that the answer here is to just update the JDs to reflect the recentish changes and ensure that API exists for plugins to deal with whatever case they want

**Comment by Toseflo** [comment]
In versions prior to 1.21.2, you could set the loot table to LootTables.EMPTY to create a mob that drops nothing. This loot table has been removed from Minecraft and Bukkit/Paper in 1.21.2. 
You can still summon a mob that doesn't drop loot by specifying an invalid loot table like /summon sheep ~ ~ ~ {DeathLootTable:""}. You can't do this with Paper because, as far as I know, Paper only allows you to select from existing loot tables. So you would have to create a datapack containing an empty loot table, which could then be selected by your plugin to create such a mob.

It would be best if Paper could restore the lost LootTables.EMPTY functionality with a built-in datapack which provides an empty loot table. This way, you could also access it with commands like /summon sheep ~ ~ ~ {DeathLootTable:"paper:empty"} or something like that.

**Comment by Doc94** [comment]
How about just add this to the LootTable interface?
```java
public static LootTable EMPTY = new LootTable() {

        @Override
        public @NotNull NamespacedKey getKey() {
            return new NamespacedKey("paper", "empty_loot_table");
        }

        @Override
        public @NotNull Collection<ItemStack> populateLoot(@Nullable final Random random, @NotNull final LootContext context) {
            return List.of();
        }

        @Override
        public void fillInventory(@NotNull final Inventory inventory, @Nullable final Random random, @NotNull final LootContext context) {

        }
    };
```

not fully like but not sure if just try to add in registry and empty work... NMS have a empty but its not registered then cannot be called in API unless mess with server side for make a thing like getEmptyLootTable for call this but the Craft implementation expect a loottable with key...
